agent/provider/src/rules.rs

```diff
@@ -239,18 +282,46 @@ impl RulesManager {
                 )
                 .map_err(|e| anyhow!("Audited-Payload rule: {e}"))?;
 
-            let mode = &self
-                .rulestore
-                .config
-                .read()
-                .unwrap()
-                .outbound
-                .audited_payload
-                .default
-                .mode;
+            let sig_cert_chain = X509Keystore::decode_cert_chain(&props.cert)?;
+
+            let rulestore_config = self.rulestore.config.read().unwrap();
+
+            // Rule set for certificate closes to leaf takes precedence
+            for sig_cert in sig_cert_chain.iter() {
+                // either:
+                // 1. a rule is set directly for the cert
+                // 2. an issuer for the certificate is in keystore and there's a rule for this cert
+                //
+                // otherwise continue to the next cert in chain
+                let rule = if let Some(rule) = rulestore_config
+                    .outbound
+                    .audited_payload
+                    .get(&cert_to_id(sig_cert)?)
+                {
+                    rule
+                } else if let Some(issuer) = self.keystore.x509_keystore().issuer(sig_cert)? {
+                    if let Some(rule) = rulestore_config.outbound.audited_payload.get(&issuer) {
+                        rule
+                    } else {
+                        continue;
+                    }
+                } else {
+                    continue;
+                };
+
+                return self
+                    .check_mode(&rule.mode, manifest)
+                    .map_err(|e| anyhow!("Audited-Payload {e}"));
+            }
```

Comment:
I would keep logic here as simple as possible and move finding issuer to the keystore itself so it presents sorted vector of cert_id's just like it is done for partner

Reply:
Moved it and created some traits to later implement Golem certs Audited Payload support